package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -786,27 +412,18 @@ cdef class FastNS(object):
             raise ValueError("Any of the box dimensions cannot be 0")
 
         self.periodic = pbc
-        self.coords = coords.astype(np.float32, order='C', copy=True)
-
-        if box.shape != (3, 3):
-            box = triclinic_vectors(box)
 
-        self.box = _PBCBox(box, self.periodic)
-
-        if cutoff < 0:
-            raise ValueError("Cutoff must be positive!")
-        if cutoff * cutoff > self.box.c_pbcbox.max_cutoff2:
-            raise ValueError("Cutoff greater than maximum cutoff ({:.3f}) given the PBC")
-
-        self.coords_bbox = self.box.fast_put_atoms_in_bbox(self.coords)
+        self.coords_bbox = distances.apply_PBC(coords, box)
```

Comment:
What if `pbc=False` and the box is too small? If a box is supplied, we should check if it is safe to use.